readthedocs/rtd_tests/tests/test_sync_versions.py

```diff
@@ -512,27 +1081,159 @@ def test_user_defined_stable_version_with_tags(self):
             ],
         }
 
-        self.client.post(
+        resp = self.client.post(
             '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
             data=json.dumps(version_post_data),
             content_type='application/json',
         )
+        self.assertEqual(resp.status_code, 200)
 
         # Didn't update to newest tag
-        version_9 = Version.objects.get(slug='0.9')
+        self.pip.refresh_from_db()
+        version_9 = self.pip.versions.get(slug='0.9')
         self.assertFalse(version_9.active)
 
         # Did update to user-defined stable version
-        version_stable = Version.objects.get(slug='stable')
+        version_stable = self.pip.versions.get(slug='stable')
         self.assertFalse(version_stable.machine)
         self.assertTrue(version_stable.active)
-        self.assertEqual('origin/stable', self.pip.get_stable_version().identifier)
+        self.assertEqual(
+            'origin/stable',
+            self.pip.get_stable_version().identifier
+        )
 
         # Check that posting again doesn't change anything from current state.
-        self.client.post(
+        resp = self.client.post(
+            '/api/v2/project/{}/sync_versions/'.format(self.pip.pk),
+            data=json.dumps(version_post_data),
+            content_type='application/json',
+        )
+        self.assertEqual(resp.status_code, 200)
+
+        self.pip.refresh_from_db()
+        version_stable = self.pip.versions.get(slug='stable')
+        self.assertFalse(version_stable.machine)
+        self.assertTrue(version_stable.active)
+        self.assertEqual(
+            'origin/stable',
+            self.pip.get_stable_version().identifier
+        )
+
+
+class TestLatestVersion(TestCase):
+    fixtures = ['eric', 'test_data']
+
+    def setUp(self):
+        self.client.login(username='eric', password='test')
+        self.pip = Project.objects.get(slug='pip')
+        Version.objects.create(
+            project=self.pip,
+            identifier='origin/master',
+            verbose_name='master',
+            active=True,
+            machine=True,
+            type=BRANCH,
+        )
+        # When the project is saved, the RTD's ``latest`` version
+        # is created.
+        self.pip.save()
+
+    def test_user_defined_latest_version_tag(self):
```

Comment:
This test is failing. When the user creates a tag named `latest`, a new version is created `latest_a`. Is that the desired behavior?

/cc @humitos

Reply:
I'm thinking about this. 

* If the user creates a `latest` and we just change our `machine` to `False`: that project looses the ability to build the latest changes in their repo. I mean, new commits to `Project.default_branch` won't trigger any build. Is that what we want?
  * this behaviour seems to be compatible with what the `stable` branch will do in a similar case

* having `latest` and `latest_a` seems to be confusing to me. Also, supposing that you want to point your readers to _your latest_: the url will be `/latest_a` which probably people won't like anyway

* I'm not sure if we have some chunk of code dependent/assuming that we will always have a `LATEST` Version with `machine=True` --that's a good checking to do before make a decision.
  * can you check this?

> For now, it seems that what we want is the first idea of just changing the `machine` field to `False`. I'd like more input here, though.

Reply:
I didn't find any query that depends on `machine=True` for latest,  but I did find some interesting relation between `latest` and `default_branch` on the code

https://github.com/rtfd/readthedocs.org/blob/b7745cd21a787c5ef14d127e34613bb3aaa2f52b/readthedocs/core/views/hooks.py#L36-L44

https://github.com/rtfd/readthedocs.org/blob/b7745cd21a787c5ef14d127e34613bb3aaa2f52b/readthedocs/projects/models.py#L792-L802

https://github.com/rtfd/readthedocs.org/blob/b7745cd21a787c5ef14d127e34613bb3aaa2f52b/readthedocs/builds/models.py#L120-L125

https://github.com/rtfd/readthedocs.org/blob/b7745cd21a787c5ef14d127e34613bb3aaa2f52b/readthedocs/projects/models.py#L315-L319

https://github.com/rtfd/readthedocs.org/blob/b7745cd21a787c5ef14d127e34613bb3aaa2f52b/readthedocs/projects/models.py#L359-L361

I think the use case of having a custom `latest` is covered by the default branch setting

![screenshot-2018-4-10 edit advanced project settings read the docs](https://user-images.githubusercontent.com/4975310/38586416-3b6a66c6-3ce3-11e8-9af1-3829550f3ba3.png)

BUT, what happens when the user wants a commit/tag as the latest version (very weird though)?

I did a quick test with a commit and a tag, nothing breaks (even if they are marked as a branch). 

So, the final question, how to handle this when a user creates a tag named `latest`? If the users want the URL to be `/latest/`, then putting the tag name on the `default_branch` setting can achieve this.

Reply:
OK, so it seems that what we want it:

* if the user defines a `latest` branch, remove the `machine=True` and handle it as a common branch
* if the user wants to trigger new builds on a non-default-latest branch, he can change the `default_branch` in the admin
* do not create a `latest_a` if the users defines a `latest` branch/tag on the repo


Those this thoughts answer your questions and unblock you to continue with this PR?

Reply:
What would happen if a user creates a `latest` tag and also touch the setting for `default_branch`? (very extreme case, I don't know if even worth considering  p:)

According to 

https://github.com/rtfd/readthedocs.org/blob/b7745cd21a787c5ef14d127e34613bb3aaa2f52b/readthedocs/projects/models.py#L315-L319

Her/his initial latest would change.

Anyway, I'm going to write more tests to check the requirements, but I think these changes are going to break others parts of the code that depends on latest, hope not.